docs/configuring.rst

```diff
@@ -8,6 +8,93 @@ contact the Globus Compute Team via Slack.
+configured to use a range of ``Providers`` which allow it to connect to Batch schedulers like Slurm and PBSTorque
+to provision compute nodes dynamically in response to workload. These capabilities
+are largely borrowed from Parsl's ``HighThroughputExecutor`` and therefore all
+of ``HighThroughputExecutor``'s parameter options are supported as passthrough.
+
+Note::
+As of ``globus-compute-endpoint==2.12.0`` ``GlobusComputeEngine`` replaces the
+``HighThroughputEngine`` as the default executor.
+
+Here are ``GlobusComputeEngine`` specific features:
+
+Retries
++++++++
+
+Functions submitted to the ``GlobusComputeEngine`` can fail due to infrastructure
+failures, for eg, the worker execution the task might terminate due to it running
+out of memory, or all workers under a batch job could fail due to the batch job
+exiting as it reaches the walltime limit. ``GlobusComputeEngine`` can be configured
+to automatically retry these tasks by setting ``max_retries_on_system_failure=N``
+where N is the number of retries allowed.
+
+Example config snippet:
+
+.. code-block:: yaml
+
+    amqp_port: 443
+    display_name: Retry_2_times
+    engine:
+        type: GlobusComputeEngine
+        max_retries_on_system_failure: 2  # Default=0
+
+
+
+Auto-Scaling
+++++++++++++
+
+``GlobusComputeEngine`` by default automatically scales workers in response to workload.
+
+Stategy configuration is limited to two options:
+1. ``max_idletime``: Maximum duration in seconds that workers are allowed to idle before they are marked for termination
+2. ``strategy_period``: Set the # of seconds between strategy attempting auto-scaling events
+
+The bounds for scaling are determined by the options to the ``Provider``
+(``init_blocks``, ``min_blocks``, ``max_blocks``). Please refer to the `https://parsl.readthedocs.io/en/stable/userguide/execution.html#elasticity <Parsl docs>`_
+for more info.
+
+Here's an example configuration:
+
+.. code-block:: yaml
+
+    engine:
+        type: GlobusComputeEngine
+
+        job_status_kwargs:
+            max_idletime: 60.0      # Default = 120s
+            strategy_period: 120.0  # Default = 5s
+
+
+Container support
++++++++++++++++++
+
+
+Containers are useful tool to deploy packaged environments to launch Functions in.
+``GlobusComputeEngine`` supports containers via launching workers in user defined
+containers. Please note that a single endpoint can only support one container type/image.
+``GlobusComputeEngine`` can launch workers in containerized environments, with
+support for Docker, Singularity, and Apptainer.
+Use by setting ``container_type``, ``container_uri`` and  additional options
+may be specified via ``container_cmd_options``.
+
+Sample configuration:
+
+  .. code-block:: yaml
+
+    display_name: Docker
+    engine:
+      type: GlobusComputeEngine
+      container_type: docker
+      container_uri: funcx/kube-endpoint:main-3.10
+      container_cmd_options: -v /tmp:/tmp
```

Comment:
Consider explaining this example in paragraph form, below.  What should the user expect this to do?

Reply:
Looking at the `container_uri`, I wonder if we ought to include any text (and examples) on how to create a container?

Reply:
I've added a line for this. Let me know if that addition makes sense.

Reply:
That's a helpful addition.  Thanks!